### PR TITLE
Add conversion to f-strings using `flynt`

### LINKED ISSUE #266: No support for using `flynt` to convert old string literal formatting to f-strings
https://github.com/ikamensh/flynt is a great tool. Think it's nice if darker includes this.

### PR DESCRIPTION
**Pull request by akaihola**
Fixes #266.

- [x] PoC implementation
- [x] unit tests for:
	- [x] `--help` content
	- [x] actual f-string conversions
- [x] documentation in README
	- [x] command line option in the `--help` copy
	- [x] description
	- [x] "New in" note
- [x] Merge #376 
- [x] Merge #377
- [x] change log

### CODE REVIEW

File: src/darker/__main__.py [hunk 1]
```diff
@@ -380,7 +380,9 @@ def main(argv: List[str] = None) -> int:
         print("\n")
 
     if args.isort and not isort:
-        raise MissingPackageError(f"{ISORT_INSTRUCTION} to use the `--isort` option.")
+        raise MissingPackageError(
+            f"{get_extra_instruction('isort')} to use the `--isort` option."
+        )
```
Comment (jedie) [inline src/darker/__main__.py:385]: What's about this kind of code for `flynt` ?!?

File: src/darker/tests/test_command_line.py [hunk 2]
```diff
@@ -469,7 +469,10 @@ def test_help_with_isort_package(capsys):
     """``darker --help`` omits ``isort`` installation instructions if it is installed"""
     with isort_present(True):
 
-        assert "Please run" not in get_darker_help_output(capsys)
+        assert (
+            "Please run `pip install darker[isort]` to enable "
+            not in get_darker_help_output(capsys)
+        )
```
Comment (jedie) [inline src/darker/tests/test_command_line.py:475]: What's about a similar code for flynt?

**Comment by akaihola** [comment]
@jedie, the implementation is still in progress. I did two preparatery refactorings which will make it simpler to add new pre-processors (like `flynt` in addition to `isort`). Would you be willing to review them as well (#376 and #377)?

**Comment by akaihola** [comment]
@jedie, there's now a proof-of-concept implementation in this branch. It sits on top of two refactorin branches, and there aren't yet any unit tests. Also, no Flynt configuration whatsoever is being taken into account – and I think that's ok for the initial version of flynt support.

Feel free to test it!

File: src/darker/__main__.py (hunk 1 above)
Comment (akaihola) [reply src/darker/__main__.py:413]: This has been now added for `flynt` as well.

File: src/darker/tests/test_command_line.py (hunk 2 above)
Comment (akaihola) [reply src/darker/tests/test_command_line.py:475]: This test has now been added for `flynt` as well.

**Comment by akaihola** [comment]
@jedie, this one is coming up in version 1.7.0. I've completed and cleaned up the branch, and marked this pull request as ready for review. Would you take a look?

There's quite a bit of clean-up and refactoring, and I tried to group things into separate commits for easier review.

**Comment by akaihola** [comment]
@Carreau, FYI, merging PR will make it easier to continue with `pyupgrade` support in #51. Would you be interested to review this one?

**Pull request merged**